runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -244,10 +236,13 @@ public long getConflictSetAndCommit(ConflictSetInfo conflictSet) {
             return NOWRITE_ADDRESS;
         }
 
-        // Write to the transaction stream if transaction logging is enabled
-        Set<UUID> affectedStreamsIds = new HashSet<>(getWriteSetInfo().getWriteSet().getEntryMap().keySet());
+        Set<UUID> affectedStreamsIds = new HashSet<>(getWriteSetInfo()
+                .getWriteSet().getEntryMap().keySet());
 
-        if (this.transaction.isLoggingEnabled()) {
+        // Write to transaction streams if transaction logging is enabled.
+        if (transaction.isLoggingEnabled()) {
+            affectedStreamsIds.addAll(getWriteSetInfo().getStreamTags());
```

Comment:
Yea, having stream tag implies requiring transaction logging. I removed that.

Reply:
@pankti-m I think we still need the add tags only if transaction logging is enabled. There needs to be a switch to turn off transction logging so that listeners won't get notifications, for example LR requires transaction logging to be turned off.